GH-33746: [R] Update NEWS.md for 11.0.0

thisisnic opened the pull request:
* Closes: #33746

thisisnic commented:
TODO: links to GH issues won't work as the pkgdown site is configured to work with JIRA (see #33631).  As a temporary fix, we could always create the URLs manually and come back to this.

ianmcook commented on r/NEWS.md:
```diff
@@ -19,6 +19,69 @@
+
+### Reading/writing data
+* new functions `open_csv_dataset()`, `open_ts_dataset()`, and 
+`open_delim_dataset()` all wrap `open_dataset()`- they don't provide new 
```
replace hyphen with em dash
```suggestion
`open_delim_dataset()` all wrap `open_dataset()`—they don't provide new 
```

ianmcook commented on r/NEWS.md:
```diff
@@ -19,6 +19,76 @@
+`dplyr::group_by()`/`dplyr::summarise()` calls are used (#14905)
+* fix for a bug in which `dplyr::summarize()` fails with division when divisor 
+is a variable (#14933)
+* fix for a bug in which as.Date fails going from timestamp[us] to timestamp[s] 
+(#14935)
```
```suggestion
* fix for a bug in which `as.Date` fails going from `timestamp[us]` to
`timestamp[s]` (#14935)
```

ianmcook commented on r/NEWS.md:
```diff
@@ -19,6 +19,76 @@
+results in an Array of length 0 (#14929)
+* fix for a bug in which accents in file paths caused an error in 
+`read_csv_arrow()` (#14930)
+* fix for a bug which prevented Arrow arrays of POSIXlt objects being created 
```
```suggestion
* fix for a bug which prevented Arrow arrays of `POSIXlt` objects being created 
```

ianmcook commented on r/NEWS.md:
```diff
@@ -19,6 +19,76 @@
+
+## Breaking changes
+
+* rlang dependency must be at least version 1.0.0 because of check_dots_empty 
```
```suggestion
* rlang dependency must be at least version 1.0.0 because of `check_dots_empty()` 
```

ianmcook commented on r/NEWS.md:
```diff
@@ -19,6 +19,76 @@
+is a variable (#14933)
+* fix for a bug in which as.Date fails going from timestamp[us] to timestamp[s] 
+(#14935)
+* fix for a bug in which Creating an Array from an object bigger than 2^31 
```
```suggestion
* fix for a bug in which creating an Array from an object bigger than 2^31 
```

nealrichardson commented:
Thanks for doing this. A couple of overall suggestions to reduce the monotony:

* There's a lot of "now" in the bullets (e.g. "can now be" over and over). "now" is implied because these are release notes, so you can generally drop it or reword if there's some contrast you're trying to make.
* Similarly, repetition of "Fix for a bug in which". I remember a recommendation somewhere about rephrasing bug fixes in a positive light, like "X correctly does Y" instead of "Fix a bug where X did Z instead of Y".

nealrichardson commented on r/NEWS.md:
```diff
@@ -19,6 +19,77 @@
+  `timestamp[s]`. (#14935)
+* Fix for a bug in which creating an Array from an object bigger than 2^31 
+  results in an Array of length 0. (#14929)
+* Fix for a bug in which accents in file paths caused an error in 
+  `read_csv_arrow()`. (#14930)
```
@paleolimbot did you mean to change the meaning of this? #14930 appears to be about paths not error messages

nealrichardson commented on r/NEWS.md:
```diff
@@ -19,6 +19,77 @@
+  results in an Array of length 0. (#14929)
+* Fix for a bug in which accents in file paths caused an error in 
+  `read_csv_arrow()`. (#14930)
+* Fix for a bug which prevented Arrow arrays of `POSIXlt` objects being created 
+  from Scalars. (#15277)
```
Here's an example of what I mean about flipping "fix for a bug in which"

```suggestion
* Arrow Scalars can be created from `POSIXlt` objects. (#15277)
```

wjones127 commented on r/NEWS.md:
```diff
@@ -19,6 +19,77 @@
+
+## Breaking changes
+
+* rlang dependency must be at least version 1.0.0 because of 
```
In general, I don't think dependency upgrades are breaking changes.

The one thing that I did find looked like a breaking change is the change in `map_batches` to be `.lazy = TRUE` by default.

thisisnic replied:
Swapped them round, thanks folks

thisisnic commented:
I've added in the URLS (this won't work automatically yet as things are configured to work with JIRA), ditched the "other" subsection and moved some items up in the hierarchy to "new feature" subsections.

eitsupi commented on r/NEWS.md:
```diff
@@ -19,6 +19,92 @@
 
 # arrow 10.0.1.9000
 
+## New features
+
+### Docs
+
+* A substantial reorganisation, rewrite of and addition to, many of the 
+  vignettes and README. (@djnavarro, 
+  [#14514](https://github.com/apache/arrow/issues/14514))  
```
Simple question: Improved documentation is new features?

thisisnic replied:
It's substantial enough to warrant inclusion, yep.

eitsupi commented on r/NEWS.md:
```diff
@@ -19,6 +19,92 @@
+  `check_dots_empty()`. (@daattali, 
+  [#14744](https://github.com/apache/arrow/issues/14744))
+
+## Breaking changes
+
+* `map_batches()` is lazy by default. ([#14521](https://github.com/apache/arrow/issues/14521))
```
Shouldn't this be at the top of the list because I think users are more interested in Breaking Changes?

thisisnic replied:
Yeah, that would mirror the tidyverse guide, so will move it

nealrichardson commented on r/NEWS.md:
```diff
@@ -19,6 +19,94 @@
+* The package can automatically link to system installations of the AWS SDK
+  for C++. (@kou, [#14235](https://github.com/apache/arrow/issues/14235))
+
+## Minor improvements and fixes
```
For consistency 

```suggestion
### Minor improvements and fixes
```

thisisnic replied:
Yeah, I was going off https://style.tidyverse.org/news.html#organisation-3

nealrichardson commented on r/NEWS.md:
```diff
@@ -19,6 +19,94 @@
+* File paths containing accents can be read by `read_csv_arrow()`. 
+  ([#14930](https://github.com/apache/arrow/issues/14930))
+
+### dplyr compatibility
```
```suggestion
### dplyr compatibility

```

nealrichardson commented on r/NEWS.md:
```diff
@@ -19,6 +19,94 @@
+  ([#15077](https://github.com/apache/arrow/issues/15077))
+* Output is accurate when multiple `dplyr::group_by()`/`dplyr::summarise()` 
+  calls are used. ([#14905](https://github.com/apache/arrow/issues/14905))
+* `dplyr::summarize()` works with division when divisor is a variable. 
```
Seems like all of these "minor improvements" up to here could go under the "dplyr compatibility", plus the one about dplyr 1.1.0 below.

thisisnic replied:
I figured these are more fixes to existing functionality than new functionality per-se, though happy to move if you think that'd make more sense?

nealrichardson commented on r/NEWS.md:
```diff
@@ -19,6 +19,94 @@
+  ([#14933](https://github.com/apache/arrow/issues/14933))
+* `as.Date()` can go from `timestamp[us]` to `timestamp[s]`. 
+  ([#14935](https://github.com/apache/arrow/issues/14935))
+* Creating an Array from an object bigger than 2^31 has correct length 
```
This could go under "Arrow object creation"

thisisnic replied:
I intentionally left it there as I figured the others were originally reported as bugs, but their resolution could be seen as feature additions, whereas this is something that was just a straight-up bug.
